### PR TITLE
Add get_auth_token

### PR DESCRIPTION
**Pull request by dri-richard**
Implements getting an auth_token as described [here](https://firebase.google.com/docs/ab-testing/abtest-config#validate_your_experiment_on_a_test_device). (The sample code is out of date so was not literally copied).

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -92,7 +92,13 @@ function init(self)
             print(err)
             return
         end
-        -- firebase is ready to use!
+		
+		-- firebase is ready to use!
```
Comment (britzl) [inline docs/index.md:96]: Nitpicking but indentation is off here!

**Review by britzl — CHANGES_REQUESTED** [review]
Don't forget to add to the script API as well:

https://github.com/defold/extension-firebase/blob/master/firebase/api/firebase.script_api

And should we not add a Gradle dependency to be explicit about the use of firebase-installations?

**Comment by dri-richard** [comment]
> And should we not add a Gradle dependency to be explicit about the use of firebase-installations?
> 
> https://firebase.google.com/docs/projects/manage-installations#android

Will do, but given that it's necessary to copy the libs manually I'm curious as to what it actually does (and how to test it)? I was also puzzled by the reference to com.google.firebase:firebase-analytics:18.0.3

**Comment by dri-richard** [comment]
Also, I now wonder if I should identifiers containing installation_auth_token instead of auth_token as there are other types of auth token in the Firebase system (e.g. authentication auth tokens)

**Comment by britzl** [comment]
> Also, I now wonder if I should use identifiers containing installation_auth_token instead of auth_token as there are other types of auth token in the Firebase system (e.g. authentication auth tokens)

Good point! Yes, I think you should as it makes it clearer what kind of token it is!



> Will do, but given that it's necessary to copy the libs manually I'm curious as to what it actually does (and how to test it)? I was also puzzled by the reference to com.google.firebase:firebase-analytics:18.0.3

It's two different things. There's the cpp part of the lib which takes care of calling the native Java code on Android. But you also need the Maven dependency which may contain Java code and Android resources.

Each release of the cpp SDK needs the correct gradle dependencies, as can be seen here:

https://raw.githubusercontent.com/firebase/firebase-cpp-sdk/v7.3.0/release_build_files/readme.md

```
Firebase Installations     | libfirebase_installations.a
                           | libfirebase_app.a
                           | com.google.firebase:firebase-installations:16.3.5
                           | (Maven package)
```

**Review by britzl — APPROVED** [review]

**Pull request merged**